apps/frontend/src/app/ErrorBoundary/index.tsx

```diff
@@ -0,0 +1,138 @@
+import { ThemeProvider } from "@emotion/react";
+import DeleteForeverIcon from "@mui/icons-material/DeleteForever"
+import ErrorIcon from "@mui/icons-material/Error";
+import RefreshIcon from "@mui/icons-material/Refresh";
+import { StyledEngineProvider, CssBaseline, Box, CardContent, Typography, Divider, CardActions, Button, Stack } from "@mui/material";
+import { Component, ErrorInfo, ReactNode } from "react";
+import { Trans, withTranslation, WithTranslation } from "react-i18next";
+import CardLight from "../Components/Card/CardLight";
+import ReadOnlyTextArea from "../Components/ReadOnlyTextArea";
+import DatabaseCard from "../PageSettings/DatabaseCard";
+import { theme } from "../Theme";
+import SpaghettiCode from "./SpaghettiCode.png";
+
+interface Props extends WithTranslation {
+  children?: ReactNode
+}
+
+interface State {
+  error?: Error
+}
+
+class ErrorBoundary extends Component<Props, State> {
```

Comment:
ew is this a class based component? what is this? 2015?

Reply:
Ikr needs to be class based because I need to implement getDerivedStateFromError for error boundary